images/virtualization-artifact/pkg/controller/kvbuilder/kvvm.go

```diff
@@ -206,24 +207,42 @@ func (b *KVVM) SetTopologySpreadConstraint(topology []corev1.TopologySpreadConst
 	b.Resource.Spec.Template.Spec.TopologySpreadConstraints = topology
 }
 
-func (b *KVVM) SetResourceRequirements(cores int, coreFraction string, memorySize resource.Quantity) error {
+func (b *KVVM) SetCpu(cores int, coreFraction string) error {
+	domainSpec := &b.Resource.Spec.Template.Spec.Domain
+	if domainSpec.CPU == nil {
+		domainSpec.CPU = &virtv1.CPU{}
+	}
 	cpuRequest, err := GetCPURequest(cores, coreFraction)
 	if err != nil {
 		return err
 	}
-	b.Resource.Spec.Template.Spec.Domain.Resources = virtv1.ResourceRequirements{
-		Requests: corev1.ResourceList{
-			corev1.ResourceCPU:    *cpuRequest,
-			corev1.ResourceMemory: memorySize,
-		},
-		Limits: corev1.ResourceList{
-			corev1.ResourceCPU:    *GetCPULimit(cores),
-			corev1.ResourceMemory: memorySize,
-		},
+	cpuLimit := GetCPULimit(cores)
+	if domainSpec.Resources.Requests == nil {
+		domainSpec.Resources.Requests = make(map[corev1.ResourceName]resource.Quantity)
 	}
+	if domainSpec.Resources.Limits == nil {
+		domainSpec.Resources.Limits = make(map[corev1.ResourceName]resource.Quantity)
+	}
+	domainSpec.Resources.Requests[corev1.ResourceCPU] = *cpuRequest
+	domainSpec.Resources.Limits[corev1.ResourceCPU] = *cpuLimit
+	domainSpec.CPU.Cores = uint32(cores)
+	domainSpec.CPU.Sockets = uint32(cpuRequest.Value())
+	domainSpec.CPU.MaxSockets = uint32(cpuLimit.Value())
```

Comment:
Set values for sockets is something new. It should be explained in PR description.